v3/src/components/web-view/use-data-interactive-controller.ts

```diff
@@ -52,53 +52,65 @@ export function useDataInteractiveController(iframeRef: React.RefObject<HTMLIFra
       webViewModel?.setDataInteractiveController(rpcEndpoint)
       webViewModel?.applyModelChange(() => {}, {log: {message: "Plugin initialized", args:{}, category: "plugin"}})
 
-      const disposer = autorun(() => {
-        const canProcessRequest = !uiState.isEditingBlockingCell
-        if (canProcessRequest && requestQueue.length > 0) {
-          uiState.captureEditingStateBeforeInterruption()
-          let tableModified = false
-          while (requestQueue.length > 0) {
-            const { request, callback } = requestQueue.nextItem
-            debugLog(DEBUG_PLUGINS, `Processing data-interactive: ${JSON.stringify(request)}`)
-            let result: DIRequestResponse = { success: false }
+      // A reaction is used here instead of an autorun so properties accessed by each handler are not
+      // observed. We only want to run the loop when a new request comes in, not when something changes
+      // that a handler accessed.
+      const disposer = reaction(() => {
+        return {
+          canProcessRequest: !uiState.isEditingBlockingCell,
+          queueLength: requestQueue.length
+        }
+      },
+      ({ canProcessRequest, queueLength }) => {
+        if (!canProcessRequest || queueLength === 0) return
+
+        uiState.captureEditingStateBeforeInterruption()
+        let tableModified = false
+
+        requestQueue.processItems(async ({ request, callback }) => {
+          debugLog(DEBUG_PLUGINS, `Processing data-interactive: ${JSON.stringify(request)}`)
+          let result: DIRequestResponse = { success: false }
 
-            const errorResult = (error: string) => ({ success: false, values: { error }} as const)
-            const processAction = (action: DIAction) => {
-              if (!action) return errorResult(t("V3.DI.Error.noAction"))
-              if (!tile) return errorResult(t("V3.DI.Error.noTile"))
+          const errorResult = (error: string) => ({ success: false, values: { error }} as const)
```

Comment:
I know this was already in the code, but this function is already defined in `di-results.ts`.